src/client/clientenvironment.cpp

```diff
@@ -47,6 +102,8 @@ ClientEnvironment::ClientEnvironment(ClientMap *map,
 	m_texturesource(texturesource),
 	m_client(client)
 {
+	auto *shdrsrc = (IWritableShaderSource *)m_client->getShaderSource();
```

Comment:
perhaps the client method should be changed to return a writable one then

Reply:
done